packages/@aws-cdk/aws-appconfig-alpha/lib/extension.ts

```diff
@@ -667,16 +678,17 @@ export class ExtensibleBase implements IExtensible {
           ],
         }),
       ],
+      name,
       ...(options?.description ? { description: options.description } : {}),
       ...(options?.latestVersionNumber ? { latestVersionNumber: options.latestVersionNumber } : {}),
-      ...(options?.name ? { name: options.name } : {}),
       ...(options?.parameters ? { parameters: options.parameters } : {}),
     });
     this.addExtensionAssociation(extension, options);
   }
 
   private addExtensionAssociation(extension: IExtension, options?: ExtensionOptions) {
-    new CfnExtensionAssociation(this.scope, `AssociationResource${this.getExtensionAssociationHash(extension)}`, {
+    const versionNumber = options?.latestVersionNumber ? options?.latestVersionNumber + 1 : 1;
+    new CfnExtensionAssociation(this.scope, `AssociationResource${this.getExtensionAssociationHash(extension.name!, versionNumber)}`, {
```

Comment:
```suggestion
    const name = extension.name ?? getExtensionDefaultName(options);
    new CfnExtensionAssociation(this.scope, `AssociationResource${this.getExtensionAssociationHash(name, versionNumber)}`, {
```
It might not be a good idea to `!` the name since this method is used also [here](https://github.com/aws/aws-cdk/blob/569593c78467404825beb277d75436bd32cdf0f9/packages/%40aws-cdk/aws-appconfig-alpha/lib/extension.ts#L654-L658).
What about creating a separate method for creating the default name for the extension to be used here and in `getExtensionForActionPoint`?
```
private getExtensionDefaultName(options?: ExtensionOptions) {
    return Names.uniqueResourceName(this.scope, {
      maxLength: 54,
      separator: '-',
    }) + '-Extension';
}